docs/javascript-overview.md

```diff
@@ -1,6 +1,90 @@
+
+## Declaring variables
+
+The recommended way to declare JavaScript variables in a notebook differs from the way it's usually done elsewhere.
+
+_**TL;DR**_ Declare your JavaScript variables without using a keyword such as `let`, `const`, or `var`, like this:
+
+```javascript
+x = 123;
+```
+
+So why is this recommended?
+
+As with other languages that weren't designed for interactive programming, using JavaScript in a notebook has a few special quirks. The difference that people most frequently encounter has to do with how to declare variables.
+
+In JavaScript, variables can be declared in a number of ways, including `let`, `var`, and `const`, as well as without a keyword.
+
+```javascript
+let declaredWithLet = 1;
+var declaredWithVar = 2;
+const declaredWithConst = 3;
+declaredWithoutKeyword = 4;
+```
+
+Since JavaScript variables are function-scoped, the three keyword-based approaches above will declare variables that can't be referenced outside of the function where they were declared. But the fourth example, declared without a keyword, will work. The following code shows this behavior:
+
+```javascript
+const doSomething = async () => {
+    let declaredWithLet = 1;
+    var declaredWithVar = 2;
+    const declaredWithConst = 3;
+    declaredWithoutKeyword = 4;
+}
+
+await doSomething();
+
+try { console.log(declaredWithLet); }          catch (e) { console.log(e.toString()); }
+try { console.log(declaredWithVar); }          catch (e) { console.log(e.toString()); }
+try { console.log(declaredWithConst); }        catch (e) { console.log(e.toString()); }
+try { console.log(declaredWithoutKeyword); }   catch (e) { console.log(e.toString()); }
+```
+
+This code produces the following output: 
+
+```console
+ReferenceError: declaredWithLet is not defined
+ReferenceError: declaredWithVar is not defined
+ReferenceError: declaredWithConst is not defined
+4
+```
+
+So why does the fourth example work? By not using the `let`, `const`, or `var` keywords with your variable declaration, you're enabling JavaScript variable hoisting to add the variable to the top-level scope. When running in a browser (including the notebook webview), this means the variable will be added to `window`. For this reason, the final line of the above example is equivalent to the following:
+
+```javascript
+console.log(window.declaredWithoutKeyword);
+```
+
+What does this have to do with Polyglot Notebooks?
+
+The Polyglot Notebooks JavaScript kernel executes your code submissions within an async arrow function, just like the above example:
+
+```javascript
+const doSomething = async () => {
+    // Your code here
```

Comment:
Just curious: Does some alternate approach exist that would allow us to execute some arbitrary piece of supplied JavaScript within a top-level scope? I am sure there is good reason to select the above approach - but was just curious if there are alternatives and what downsides these alternatives might have.